File: app/src/main/java/com/edorin/kotlindemo/java/SummaryActivity.java
package com.edorin.kotlindemo.java;

import android.annotation.SuppressLint;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import com.edorin.kotlindemo.R;

public class SummaryActivity extends AppCompatActivity {

    TextView txvName, txvGender;

    @SuppressLint("SetTextI18n")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_summary);

        // initialize widgets
        txvName = findViewById(R.id.txvName);
        txvGender = findViewById(R.id.txvGender);

        // get intent extras
        String firstname = getIntent().getStringExtra("firstname");
        String lastname = getIntent().getStringExtra("lastname");
        String gender = getIntent().getStringExtra("gender");

        // set texts
        txvName.setText(lastname + ", " + firstname);
        txvGender.setText(gender);
    }

}

File: app/src/main/java/com/edorin/kotlindemo/kotlin/SummaryActivity.kt
package com.edorin.kotlindemo.kotlin

import android.annotation.SuppressLint
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import com.edorin.kotlindemo.R
import com.edorin.kotlindemo.stringExtra
import kotlinx.android.synthetic.main.activity_summary.*

class SummaryActivity : AppCompatActivity() {

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_summary)

        // get string extras
        val firstname = "firstname".stringExtra(this)
        val lastname = "lastname".stringExtra(this)
        val gender = "gender".stringExtra(this)

        // set texts
        txvName.text = "$lastname, $firstname"
        txvGender.text = gender
    }

}